src/openms/source/ANALYSIS/ID/AScore.cpp

```diff
@@ -360,23 +396,29 @@ namespace OpenMS
       suf_with_phospho_first = AASequence::fromString(first.getSuffix(first.size() - candidates.second).toString());
       suf_with_phospho_second = AASequence::fromString(second.getSuffix(first.size() - candidates.second).toString());
     }
+
     spectrum_generator.addPeaks(prefix, pref, Residue::BIon, charge);
     spectrum_generator.addPeaks(suffix, suf, Residue::YIon, charge);
     spectrum_generator.addPeaks(prefix_with_phospho_first, pref_with_phospho_first, Residue::BIon, charge);
     spectrum_generator.addPeaks(prefix_with_phospho_second, pref_with_phospho_second, Residue::BIon, charge);
     spectrum_generator.addPeaks(suffix_with_phospho_first, suf_with_phospho_first, Residue::YIon, charge);
     spectrum_generator.addPeaks(suffix_with_phospho_second, suf_with_phospho_second, Residue::YIon, charge);
```

Comment:
I agree, the charge here should be one for most if not all cases and *not* the precursor charge, that makes no sense whatsoever. There may be cases where it makes sense to have 1+ and 2+ fragments but these are very few